src/hotspot/share/gc/shared/gcLocker.hpp

```diff
@@ -27,126 +27,30 @@
 
 #include "gc/shared/gcCause.hpp"
 #include "memory/allStatic.hpp"
-#include "utilities/globalDefinitions.hpp"
-#include "utilities/macros.hpp"
-
-class JavaThread;
-
-// The direct lock/unlock calls do not force a collection if an unlock
-// decrements the count to zero. Avoid calling these if at all possible.
+#include "runtime/mutex.hpp"
 
 class GCLocker: public AllStatic {
- private:
-  // The _jni_lock_count keeps track of the number of threads that are
-  // currently in a critical region.  It's only kept up to date when
-  // _needs_gc is true.  The current value is computed during
-  // safepointing and decremented during the slow path of GCLocker
-  // unlocking.
-  static volatile jint _jni_lock_count;  // number of jni active instances.
-  static volatile bool _needs_gc;        // heap is filling, we need a GC
-  static uint _total_collections;        // value for _gc_locker collection
+  static Monitor* _lock;
```

Comment:
Not sure if having this copy/reference to `Heap_lock` makes the code more clear than referencing `Heap_lock` directly. It needs to be `Heap_lock` anyway.